### DISCUSSION TITLE
How do I handle the reset password process in React?

### DISCUSSION THREAD

**Discussion by DKBoston15** [supabase/supabase#1920] [Questions] [answered]
I am building out an authentication flow and working on the reset password process, but I cannot make sense of the documentation. This is entirely due to my lack of knowledge here, but I cannot find a single good example anywhere. Could someone explain to me how I am supposed to do what is in the docs in React? I've got the function setup to send the email and redirect to a reset-password page. However, I'm not sure how to capture the information I need. Additionally, when my page loads, I end up with a url like this `localhost:3000/#`. All of the information shown in the docs is shown briefly but disappears.

<img width="965" alt="Screen Shot 2021-06-10 at 10 48 24 PM" src="https://user-images.githubusercontent.com/46634008/121627962-f8262f80-ca3d-11eb-9a0e-13075d8ff7e2.png">

Here is my function for sending the password reset email.
```
        resetPassword: (data) =>
            supabase.auth.api.resetPasswordForEmail(data, {
                redirectTo: `http://localhost:3000/reset-password`
            })
```

Here is the page it redirects to so far. (Note: I haven't setup the form yet, just trying to capture the access token).
```
export default function PasswordReset() {
    let history = useHistory()
    useEffect(() => {
        const parsed = queryString.parse(window.location.search)
        if (parsed.type === "recovery") {
            return
        }
        history.push("/sign-in")
    })

    return (
        <div>
            <h1>Reset Password Here!</h1>
        </div>
    )
}
```

**Comment by easy-street** [comment, marked as answer]
the [Supabase UI](https://ui.supabase.com/) `Auth` component can handle some of the heavy lifting.
![image](https://user-images.githubusercontent.com/64047794/121794630-9c8bab80-cbcf-11eb-902e-e82baa1b6778.png)

in your app:
- handle the `PASSWORD_RECOVERY` and `USER_UPDATED` events in the `supabase.auth.onAuthStateChange` callback, setting a local state value to indicate the view to render
- when the `PASSWORD_RECOVERY` event is received, render the `Auth.UpdatePassword` component

examples:
- the [nextjs-with-supabase-auth example](https://github.com/supabase/supabase/blob/master/examples/nextjs-with-supabase-auth/pages/index.js#L21) implements password recovery with the Supabase UI `Auth` component
- [ported to React/CRA here](https://github.com/easy-street/supabase/blob/feature/add-react-with-supabase-auth-example/examples/react-with-supabase-auth/src/App.js)

**Reply by DKBoston15** [reply]
Cheers, the ported example did the trick!
